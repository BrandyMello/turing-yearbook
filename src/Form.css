.form {
  display: inline-grid;
}

input {
  height: 30px;
  width: auto;
  margin-bottom: 10px;
}

.submit {
  background-color: #ff63479e;
  border-radius: 50%;
  height: 30px;
  margin: 10px;
}