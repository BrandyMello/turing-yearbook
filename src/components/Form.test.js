import React from 'react';
import { shallow } from 'enzyme';
import Form from './StudentForm';

describe('Form', () =>  {
  it('should match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow(<Form addCard={jest.fn()}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should change the state on the input change within the form', () => {
    const wrapper = shallow(<Form addCard={jest.fn()}/>);

    wrapper.find('input').at(0).simulate('change', {target: {value: 'Berry', name: 'name'}});
    wrapper.find('input').at(1).simulate('change', {target: {value: 'trying', name: 'quote'}});
    wrapper.find('input').at(2).simulate('change', {target: {value: 'carry on', name: 'supperlative'}});

    expect(wrapper.state('name')).toEqual('Berry');
    expect(wrapper.state('quote')).toEqual('trying');
    expect(wrapper.state('supperlative')).toEqual('carry on');
  });

  it('should call addCard prop when clicked', () => {
    const addStudentMock = jest.fn();
    const wrapper = shallow(<Form addStudent={jest.fn()}/>);
    const eventMock = { preventDefault: jest.fn() };

    wrapper.find('button').simulate('click', eventMock);
    wrapper.instance().handleNewStudent(eventMock);

    expect(handleNewStudent).toHaveBeenCalled();
  });

});