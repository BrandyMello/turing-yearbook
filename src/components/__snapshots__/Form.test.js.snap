// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all the data passed correctly 1`] = `
<form
  className="form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Type name here: \\"Jane\\""
    type="text"
    value=""
  />
  <input
    name="quote"
    onChange={[Function]}
    placeholder="Type fave quote here..."
    type="text"
    value=""
  />
  <input
    name="superlative"
    onChange={[Function]}
    placeholder="Type superlative here..."
    type="text"
    value=""
  />
  <button
    className="submit"
    onClick={[Function]}
  >
    Make me a Student
  </button>
</form>
`;
